internal/chain/reward.go

```diff
@@ -201,53 +283,53 @@ func AccumulateRewards(
 			// Finally do the pay
 			for bucket := range resultsHandle {
 				for payThem := range resultsHandle[bucket] {
-					snapshot, err := bc.ReadValidatorSnapshot(resultsHandle[bucket][payThem].payee)
+					snapshot, err := bc.ReadValidatorSnapshot(
+						resultsHandle[bucket][payThem].EcdsaAddress,
+					)
 					if err != nil {
-						return network.NoReward, err
+						return network.EmptyPayout, err
 					}
-					due := resultsHandle[bucket][payThem].payout.TruncateInt()
-					newRewards = new(big.Int).Add(newRewards, due)
+					due := resultsHandle[bucket][payThem].payout
+					newRewards.Add(newRewards, due)
 					if err := state.AddReward(snapshot, due); err != nil {
-						return network.NoReward, err
+						return network.EmptyPayout, err
 					}
 				}
 			}
 
-			return newRewards, nil
+			return network.NewStakingEraRewardForRound(newRewards, missing), nil
 		}
-		return network.NoReward, nil
+		return network.EmptyPayout, nil
 	}
 
 	// Before staking
-	payable := []struct {
-		string
-		common.Address
-		*big.Int
-	}{}
-
 	parentHeader := bc.GetHeaderByHash(header.ParentHash())
 	if parentHeader.Number().Cmp(common.Big0) == 0 {
 		// Parent is an epoch block,
 		// which is not signed in the usual manner therefore rewards nothing.
-		return network.NoReward, nil
+		return network.EmptyPayout, nil
 	}
 
 	_, signers, _, err := availability.BallotResult(bc, header, header.ShardID())
 
 	if err != nil {
-		return network.NoReward, err
+		return network.EmptyPayout, err
 	}
 
-	totalAmount := rewarder.Award(
-		network.BlockReward, signers, func(receipient common.Address, amount *big.Int) {
-			payable = append(payable, struct {
-				string
-				common.Address
-				*big.Int
-			}{common2.MustAddressToBech32(receipient), receipient, amount},
-			)
-		},
-	)
+	totalAmount := big.NewInt(0)
+
+	{
+		last := big.NewInt(0)
```

Comment:
This is just copy and paste from the old code right?

Reply:
yes, see from that deleted Award, itself a copy paste from orginal code